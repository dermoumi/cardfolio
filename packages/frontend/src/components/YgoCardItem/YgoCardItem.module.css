.ygo-card-item {
  aspect-ratio: 169 / 246;
  display: inline-block;
  background-color: #f0f0f0;
  border-radius: 0.4cqw;
  overflow: hidden;

  & > img {
    width: 100%;
    height: 100%;
  }
}
