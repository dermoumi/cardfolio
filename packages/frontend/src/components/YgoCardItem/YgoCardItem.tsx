import type { YgoCard } from "@/types";
import type { FC } from "react";

import styles from "./YgoCardItem.module.css";

export type YgoCardItemProps = {
  card: YgoCard;
};

const YgoCardItem: FC<YgoCardItemProps> = ({ card }) => {
  const imgSrc = `/api/v1/ygo/cards/${card.id}/image`;

  return (
    <div className={styles.ygoCardItem}>
      <img src={imgSrc} alt={card.name} />
    </div>
  );
};

export default YgoCardItem;
