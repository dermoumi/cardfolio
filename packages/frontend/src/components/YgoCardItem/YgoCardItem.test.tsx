import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import YgoCardItem from ".";

describe("YgoCardItem", () => {
  const mockCard = {
    "id": 10,
    "updatedAt": "2025-09-03T21:54:02.391850Z",
    "name": "1st Movement Solo",
    "description":
      "If you control no monsters: Special Summon 1 Level 4 or lower \"Melodious\" monster from your hand or Deck. You can only activate 1 \"1st Movement Solo\" per turn. You cannot Special Summon monsters during the turn you activate this card, except \"Melodious\" monsters.",
    "kind": "spell",
    "password": "44256816",
    "konamiId": 11391,
    "tcgDate": "2014-11-06",
    "ocgDate": "2014-07-19",
    "spellKind": "normal",
  } as const;

  it("renders card image with correct src and alt", () => {
    render(<YgoCardItem card={mockCard} />);
    const imgElement: HTMLImageElement = screen.getByAltText("1st Movement Solo");
    expect(imgElement.src).toContain("/api/v1/ygo/cards/10/image");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<YgoCardItem card={mockCard} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
