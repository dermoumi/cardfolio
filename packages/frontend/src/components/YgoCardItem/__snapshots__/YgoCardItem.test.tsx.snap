// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YgoCardItem > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_ygoCardItem_b4a956"
  >
    <img
      alt="1st Movement Solo"
      src="/api/v1/ygo/cards/10/image"
    />
  </div>
</DocumentFragment>
`;
